src/main/java/duke/Command.java

```diff
@@ -0,0 +1,23 @@
+package duke;
+
+/**
+ * Represents an executable command.
+ */
+public abstract class Command {
+
+    /**
+     * Executes the command.
+     *
+     * @param tasks tasks in the task list.
+     * @param ui ui that interacts with users.
+     * @param storage storage to save tasks.
+     * @throws Exception  if there is an error.
+     */
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws Exception;
+
+    /**
+     * Checks if command is exit command
+     * @return true if it is exit command
+     */
+    public abstract boolean isExit();
```

Comment:
Not sure if this is the right approach, considering how there is only one exit command (and almost all of the commands will return false)